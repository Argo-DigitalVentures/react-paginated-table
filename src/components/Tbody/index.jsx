import React, { PropTypes } from 'react';

import TrComponent from '../Tr';
import TdComponent from '../Td';

import {
  styleType,
  classNameType,
  dataType,
  columnType,
} from '../../propTypes';

function Tbody(props) {
  const {
    style,
    className,
    data,
    columns,
    TrComponent,
    TdComponent,
    trStyle,
    tdStyle,
    trClassName,
    tdClassName,
    ...rest
  } = props;
  return (
    <tbody style={style} className={className} {...rest}>
      {
        data.map((row, rowIndex, data) => (
          <TrComponent
            key={rowIndex} // eslint-disable-line react/no-array-index-key
            style={trStyle}
            className={trClassName}
            columns={columns}
            row={row}
            rowIndex={rowIndex}
            data={data}
            CellComponent={TdComponent}
            cellStyle={tdStyle}
            cellClassName={tdClassName}
          />
        ))
      }
    </tbody>
  );
}

Tbody.defaultProps = {
  style: {},
  className: '',
  TrComponent,
  TdComponent,
  trStyle: {},
  tdStyle: {},
  trClassName: '',
  tdClassName: '',
};

Tbody.propTypes = {
  data: dataType.isRequired,
  columns: PropTypes.arrayOf(columnType).isRequired,
  style: PropTypes.object, // eslint-disable-line react/forbid-prop-types
  className: PropTypes.string,
  TrComponent: PropTypes.func,
  TdComponent: PropTypes.func,
  trStyle: styleType,
  tdStyle: styleType,
  trClassName: classNameType,
  tdClassName: classNameType,
};

export default Tbody;
