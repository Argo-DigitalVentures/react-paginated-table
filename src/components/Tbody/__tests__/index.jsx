import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Tbody from '../';

const data = [
  { company: 'Argo', location: 'New York' },
  { company: 'John Hancock', location: 'New York' },
];

const columns = [
  { id: 'company', header: 'Company Name', selector: 'company' },
  { id: 'location', header: () => null, selector: 'location' },
];

describe('<Tbody />', () => {
  it('should accept className, style, and other DOM props', () => {
    const tree = shallow(
      <Tbody
        style={{
          color: 'green',
        }}
        className="test"
        data={data}
        columns={columns}
      />,
    );
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('should use a custom row and cell components', () => {
    const FakeTrComponent = () => null;
    const FakeTdComponent = () => null;
    const tree = shallow(
      <Tbody
        style={{
          color: 'green',
        }}
        className="test"
        data={data}
        columns={columns}
        TrComponent={FakeTrComponent}
        TdComponent={FakeTdComponent}
        trStyle={{
          color: 'yellow',
        }}
        tdStyle={{
          color: 'purple',
        }}
        trClassName="trTestClassName"
        tdClassName="tdTestClassName"
      />,
    );
    expect(toJson(tree)).toMatchSnapshot();
  });
});
