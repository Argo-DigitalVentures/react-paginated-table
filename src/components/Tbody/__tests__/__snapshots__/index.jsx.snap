exports[`<Tbody /> should accept className, style, and other DOM props 1`] = `
<tbody
  className="test"
  style={
    Object {
      "color": "green",
    }
  }>
  <Tr
    CellComponent={[Function]}
    cellClassName=""
    cellStyle={Object {}}
    className=""
    columns={
      Array [
        Object {
          "header": "Company Name",
          "id": "company",
          "selector": "company",
        },
        Object {
          "header": [Function],
          "id": "location",
          "selector": "location",
        },
      ]
    }
    data={
      Array [
        Object {
          "company": "Argo",
          "location": "New York",
        },
        Object {
          "company": "John Hancock",
          "location": "New York",
        },
      ]
    }
    row={
      Object {
        "company": "Argo",
        "location": "New York",
      }
    }
    rowIndex={0}
    style={Object {}} />
  <Tr
    CellComponent={[Function]}
    cellClassName=""
    cellStyle={Object {}}
    className=""
    columns={
      Array [
        Object {
          "header": "Company Name",
          "id": "company",
          "selector": "company",
        },
        Object {
          "header": [Function],
          "id": "location",
          "selector": "location",
        },
      ]
    }
    data={
      Array [
        Object {
          "company": "Argo",
          "location": "New York",
        },
        Object {
          "company": "John Hancock",
          "location": "New York",
        },
      ]
    }
    row={
      Object {
        "company": "John Hancock",
        "location": "New York",
      }
    }
    rowIndex={1}
    style={Object {}} />
</tbody>
`;

exports[`<Tbody /> should use a custom row and cell components 1`] = `
<tbody
  className="test"
  style={
    Object {
      "color": "green",
    }
  }>
  <FakeTrComponent
    CellComponent={[Function]}
    cellClassName="tdTestClassName"
    cellStyle={
      Object {
        "color": "purple",
      }
    }
    className="trTestClassName"
    columns={
      Array [
        Object {
          "header": "Company Name",
          "id": "company",
          "selector": "company",
        },
        Object {
          "header": [Function],
          "id": "location",
          "selector": "location",
        },
      ]
    }
    data={
      Array [
        Object {
          "company": "Argo",
          "location": "New York",
        },
        Object {
          "company": "John Hancock",
          "location": "New York",
        },
      ]
    }
    row={
      Object {
        "company": "Argo",
        "location": "New York",
      }
    }
    rowIndex={0}
    style={
      Object {
        "color": "yellow",
      }
    } />
  <FakeTrComponent
    CellComponent={[Function]}
    cellClassName="tdTestClassName"
    cellStyle={
      Object {
        "color": "purple",
      }
    }
    className="trTestClassName"
    columns={
      Array [
        Object {
          "header": "Company Name",
          "id": "company",
          "selector": "company",
        },
        Object {
          "header": [Function],
          "id": "location",
          "selector": "location",
        },
      ]
    }
    data={
      Array [
        Object {
          "company": "Argo",
          "location": "New York",
        },
        Object {
          "company": "John Hancock",
          "location": "New York",
        },
      ]
    }
    row={
      Object {
        "company": "John Hancock",
        "location": "New York",
      }
    }
    rowIndex={1}
    style={
      Object {
        "color": "yellow",
      }
    } />
</tbody>
`;
